src/main.c

```diff
@@ -1306,6 +1326,13 @@ static void on_sigfatal(int signo)
     int framecnt = backtrace(frames, sizeof(frames) / sizeof(frames[0]));
     backtrace_symbols_fd(frames, framecnt, crash_handler_fd);
 
+    if (conf.crash_handler_wait_pipe_close) {
+        struct pollfd pfd[1];
+        pfd[0].fd = crash_handler_fd;
+        pfd[0].events = POLLERR | POLLHUP;
+        poll(pfd, 1, -1);
```

Comment:
It might be wiser to check for EINTR (and call poll again in such case), considering the fact that signals other than the one that is being handled by this signal handler may preempt the execution of `poll`.